### PR TITLE
Fix comment in `poly_project_and_unify_type`

### PR DESCRIPTION
**Pull request by compiler-errors**
Renamed some variants to match `ProjectAndUnifyResult`

### CODE REVIEW

File: compiler/rustc_trait_selection/src/traits/project.rs [hunk 1]
```diff
@@ -164,18 +164,19 @@ pub(super) enum ProjectAndUnifyResult<'tcx> {
...
+/// - `MismatchedProjectionTypes`: the projection can be normalized, but
+///   is not equal to the expected type.
+/// - `Recursive`: this is called recursively while normalizing the same
+///   projection.
+/// - `FailedNormalization`: The projection cannot be normalized due to
+///   ambiguity (resolving some inference variables in the projection may
+///   fix this).
+/// - `Holds(obligations)`: The projection bound holds subject to the given
+///   obligations. If the projection cannot be normalized because the
+///   required trait bound doesn't hold this returned with `obligations`
```
Comment (lcnr) [inline compiler/rustc_trait_selection/src/traits/project.rs:178]:
```suggestion
///   required trait bound doesn't hold this is returned with `obligations`
```

File: compiler/rustc_trait_selection/src/traits/project.rs [hunk 2]
```diff
@@ -164,18 +164,19 @@ pub(super) enum ProjectAndUnifyResult<'tcx> {
 /// If successful, this may result in additional obligations. Also returns
 /// the projection cache key used to track these additional obligations.
 ///
-/// ## Returns
+/// ## Returns (`ProjectAndUnifyResult`)
```
Comment (lcnr) [inline compiler/rustc_trait_selection/src/traits/project.rs:167]: Can you instead move that comment to the type definition of `ProjectAndUnifyResult`?

Comment (compiler-errors) [reply compiler/rustc_trait_selection/src/traits/project.rs]: Done.

**Pull request merged**